code/controllers/master.dm

```diff
@@ -63,6 +67,12 @@ var/CURRENT_TICKLIMIT = TICK_LIMIT_RUNNING
 		Master = src
 
 /datum/controller/master/Destroy()
+	running = FALSE
+	//Give the loop thread some shutdown time
+	while(looping)
+		sleep(1)
```

Comment:
this isn't needed because the mc's loop never sleeps outside of being finished, it uses all waitfor procs so nothing it calls can sleep, ie, the only time the mc will still be looping is if a proc called by the mc tried to delete it.